### ISSUE TITLE
[Bug] Fetch challenge mode time left being nil causing errors in menu

### ISSUE THREAD

**Issue by YuLun-bili** [Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker#440]
### Describe the bug

as described in title

### Steps to reproduce the behavior

1. make sure a challenge level's time left would be read as `nil` in game (I don't know how this happened, possibly because I did that challenge way earlier back in history and game simply didn't record time)
2. open challenge mode menu then select the level
4. `[string "data/script/challenge.lua"]:32: bad argument #2 to 'format' (number expected, got nil)`

https://github.com/Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker/assets/70589524/af9ccd49-ec42-4150-9f0d-e8721e06144c

### Expected behavior

no errors

### Environment

- Teardown Version 1.5.1

### Additional context

savegame.xml as follows:
```xml
<challenge>
	<ch_marina_hunted>
		<score value="6.0"/>
		<stars value="1"/>
		<scoredetails value="6 chests"/>
	</ch_marina_hunted>
	<ch_marina_mayhem>
		<score value="1241886.0"/>
		<stars value="5"/>
		<scoredetails value="1241886 voxels"/>
	</ch_marina_mayhem>
	<ch_lee_hunted>
		<score value="19.0"/>
		<stars value="3"/>
		<scoredetails value="19 chests"/>
	</ch_lee_hunted>
	<ch_lee_mayhem>
		<score value="561399.0"/>
		<stars value="2"/>
		<scoredetails value="561399 voxels"/>
	</ch_lee_mayhem>
	<ch_mansion_hunted>
		<score value="26.0"/>
		<stars value="5"/>
		<scoredetails value="26 chests"/>
	</ch_mansion_hunted>
	<ch_mall_hunted>
		<score value="25.0"/>
		<stars value="5"/>
		<scoredetails value="25 chests"/>
	</ch_mall_hunted>
	<ch_caveisland_hunted>
		<score value="9.0"/>
		<stars value="1"/>
		<scoredetails value="9 chests"/>
	</ch_caveisland_hunted>
	<ch_mansion_fetch>
		<score value="8.0"/>
		<stars value="4"/>
		<scoredetails value="8 targets"/>
	</ch_mansion_fetch>
	<ch_factory_hunted>
		<score value="7.0"/>
		<stars value="1"/>
		<scoredetails value="7 chests"/>
	</ch_factory_hunted>
</challenge>
```

**Comment by YuLun-bili** [comment]
update, using arrow keys to browse the menu also cause this exact same bug

**Comment by YuLun-bili** [comment]
fix as simple as a `nil` check when passing the value, or make this slightly better such as this:
```lua
local timeLeftString = GetTranslatedStringByKey("NAME_UNKNOWN")
if timeLeft then timeLeftString = string.format("%.1f", timeLeft) end
scoreDetailsString = GetTranslatedStringByKey("TXT_SCORE_TARGETS," .. math.floor(score)) .."\n" .. GetTranslatedStringByKey("TXT_SCORE_SECONDS_LEFT," .. timeLeftString)
```

**Comment by YuLun-bili** [comment]
updated original report

**Comment by Gregory-Gregory** [comment]
Hi @YuLun-bili,

To avoid misunderstanding, does this [solution ](https://github.com/Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker/issues/440#issuecomment-1832395581) solve the issue? Or do you still have it with different options?

Thank you!

**Comment by YuLun-bili** [comment]
> Hi @YuLun-bili,
> 
> To avoid misunderstanding, does this [solution ](https://github.com/Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker/issues/440#issuecomment-1832395581) solve the issue? Or do you still have it with different options?
> 
> Thank you!

It at least won't throw errors. I just proposed a way to display "unknown" if time is not recorded, dev team could do whatever changes they like to.

**Comment by Gregory-Gregory** [comment]
@YuLun-bili thank you for the clarification.

Just in case, I shared this information with the team.

**Comment by YuLun-bili** [comment]
hmmm, error still there but as a result of #531 it's not showing errors on screen. But it still errors and terminates function called and omits "Play" button
<img src="https://github.com/Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker/assets/70589524/c77c82be-343a-4412-b038-37e559609ea3" width="600px">

**Comment by YuLun-bili** [comment]
resolved in 1.6.0
